operator-common/src/main/java/io/strimzi/operator/common/operator/resource/SecretOperator.java

```diff
@@ -29,4 +36,28 @@ public SecretOperator(Vertx vertx, KubernetesClient client) {
     protected MixedOperation<Secret, SecretList, Resource<Secret>> operation() {
         return client.secrets();
     }
+
+
+    public Future<String> getCertificateAsync(String namespace, CertSecretSource certSecretSource) {
+        return this.getAsync(namespace, certSecretSource.getSecretName())
+                .compose(secret -> Future.succeededFuture(secret.getData().get(certSecretSource.getCertificate())));
+    }
+
+    public Future<CertAndKey> getCertificateAndKeyAsync(String namespace, KafkaClientAuthenticationTls auth) {
+        return this.getAsync(namespace, auth.getCertificateAndKey().getSecretName())
+                .compose(secret -> Future.succeededFuture(new CertAndKey(secret.getData().get(auth.getCertificateAndKey().getKey()).getBytes(), secret.getData().get(auth.getCertificateAndKey().getCertificate()).getBytes())));
+    }
+
+    public Future<String> getPasswordAsync(String namespace, KafkaClientAuthentication auth) {
+        if (auth instanceof KafkaClientAuthenticationPlain) {
+            return this.getAsync(namespace, ((KafkaClientAuthenticationPlain) auth).getPasswordSecret().getSecretName())
+                    .compose(secret -> Future.succeededFuture(secret.getData().get(((KafkaClientAuthenticationPlain) auth).getPasswordSecret().getPassword())));
+        }
+        if (auth instanceof KafkaClientAuthenticationScramSha512) {
+            return this.getAsync(namespace, ((KafkaClientAuthenticationScramSha512) auth).getPasswordSecret().getSecretName())
+                    .compose(secret -> Future.succeededFuture(secret.getData().get(((KafkaClientAuthenticationScramSha512) auth).getPasswordSecret().getPassword())));
+        } else {
+            return Future.failedFuture("Auth type " + auth.getType() + " does not have a password property");
+        }
+    }
```

Comment:
Do we really want to have these here? It introduces the `api` module into this quite heavily which was not the case before. They are also very specific where as the SecretOperator purpose is very generic.

Reply:
@tombentley Do we want to pass certificate, key, and password as a String parameter?

Reply:
The `api` is just part of it ... what is the value of having this here instead of in the Util class?

Reply:
Currently we've failed to really encapsulate and abstract our certificate handling. What I mean by that is everywhere we pass around a `Secret` which we _happen to know_ contains a key store (broker certificate and key), or truststore (ca certificate). But that requires that the consumer of that `Secret` knows what _type_ of secret is it, and how to decode the values within it into something it can use. 

I think there is value is trying to better encapsulate these concepts, so rather than passing around Secrets we pass around something higher level. For example a `AdminClientSupplier` might accept some Strimzi `Truststore` class and a Strimzi `Keystore` class. Providing Resource Operators for those things (`Truststore` and `Keystore`) would be a logical thing to do. This would provide a path towards being able to abstract where certain credentials are stored (e.g. option to store keys in Vault), and should make the KAO slightly simpler to understand, because there are fewer generic Secrets floating around.

At least that was my motivation for suggesting this, and I'd be interested to know whether people think that's a desirable goal or not (and if they have specific reasons to think it wouldn't work out).